### PR TITLE
Migrate taleo to greenhouse

### LINKED ISSUE #220: migrate all-vacanies page from taleo to greehouse.io for job board
## Summary

using the greenhouse.io api, we need to update the /careers/all-vacancies page to use greenhouse.io instead of taleo

### PR DESCRIPTION
**Pull request by pmahnke**
## Done

* HR didn't want to use the greenhouse.io job listing page as the departments were incorrect, so using the api, we have replaced the older taleo page that will link to the hosted greenhouse job descriptions
* Robin added a new get_raw_json_feed template tag to read this format of json

## QA

- download this branch and `./run` the server
- look at http://0.0.0.0:8002/careers/all-vacancies and see that the page lists all open jobs, grouped by an external department field from greenhouse.io

## Issue / Card

Fixes #220 

## Screenshots

image

### CODE REVIEW

**Comment by pmahnke** [comment]
this will go live on 11 December 2017

**Comment by kwm14** [comment]
## Visual review

Few visual tweaks :slightly_smiling_face:

### Summary page
- Update hero section to use `p-strip--image is-dark` class and inherit aubergine background as documented [here](https://docs.vanillaframework.io/en/patterns/strip#strip-image)
- Update department titles to use `p-heading--four`
- Update margin-top on Sales department to `16px` so it's consistent with other department styling
- Visual reference below with my quick hack

screen shot 2017-11-24 at 14 34 29

### Job description page
- The styling me and Caleb implemented has changed? Looks as if it's reverted back to original styling before we made design tweaks to it...

File: templates/careers/all-vacancies.html [hunk 1]
```diff
@@ -1,20 +1,56 @@
...
 {% block content %}
-
 <section class="p-strip--light">
   <div class="row">
     <div class="col-8">
       <h1>All current vacancies</h1>
-      <p>If you are interested in a role in a specific department or location, you can <a href="https://ldd.tbe.taleo.net/ldd03/ats/careers/jobSearch.jsp?org=CANONICAL&amp;cws=1&amp;rid=86">search jobs</a>.</p>
-      {% get_rss_feed "http://ldd.tbe.taleo.net/ldd03/ats/servlet/Rss?org=CANONICAL&_rss_version=2&cws=1" as vacancies %} {% include "_includes/vacancies.html" with vacancies=vacancies %}
+      {% if jobs is False %}
+        <p class="u-no-margin--bottom">Feed failed to load. If you have time, please <a href="https://github.com/canonical-websites/www.canonical.com/issues/new">file an issue</a>.</p>
```
Comment (anthonydillon) [inline templates/careers/all-vacancies.html]: Maybe use an error notification?

File: templates/careers/job.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+{% block content %}
+
+<section class="p-strip--light">
+  <div class="row">
+    <div class="col-8">
+      <h1>All current vacancies</h1>
+      {% get_raw_json_feed "https://api.greenhouse.io/v1/boards/Canonical/jobs/"|add:job_id  as listing %}
+
+{% if listing is False %}
+<p>got nothing</p>
```
Comment (anthonydillon) [inline templates/careers/job.html]: Descriptive notification?

File: templates/careers/job.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+{% else %}
+{% autoescape off %}
+{{ listing.content |decode }}
```
Comment (anthonydillon) [inline templates/careers/job.html]: Remove the space before the pipe

File: templates/careers/job.html [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+{% block content %}
+
+<section class="p-strip--light">
+  <div class="row">
+    <div class="col-8">
+      <h1>All current vacancies</h1>
+      {% get_raw_json_feed "https://api.greenhouse.io/v1/boards/Canonical/jobs/"|add:job_id  as listing %}
+
+{% if listing is False %}
```
Comment (anthonydillon) [inline templates/careers/job.html]: Can you indent please

**Comment by pmahnke** [comment]
@kwm14 updated, however used p-strip--accent as --image had no background?

@anthonydillon removed job.html page as it is not needed ... added the error notification

image

**Comment by anthonydillon** [comment]
Cheers @pmahnke, I think the notification should be on the strip below the hero strip. Where the feed is outputted.

**Comment by pmahnke** [comment]
@anthonydillon I knew you were going to say that... more logic!

**Comment by pmahnke** [comment]
@anthonydillon 

* moved the notification of no results into the white content area
* added a check if the feed failed to return results

image

**Comment by kwm14** [comment]
@pmahnke looking good just one small update you missed:

- [ ] Update margin-top on Sales department to `16px` so it's consistent with other department styling

**Comment by kwm14** [comment]
Thanks @pmahnke 👍

**Comment by anthonydillon** [comment]
This branch could to merged into a feature branch and create a go live card for next iteration.

**Pull request merged**